tests/test_zone_translation.py

```diff
@@ -0,0 +1,382 @@
+    zone_2.to_file(main_dir / "zone_2.shp")
+    return file
+
+
+@pytest.fixture(scope="class")
+def zone_2_moved(zone_2_shape, main_dir):
+    gdf = gpd.read_file(zone_2_shape)
+    gdf.affine_transform(1, 0, 0, 1, 4, 4)
+    file = main_dir / 'zone_2_altered.shp'
+    gdf.to_file(file)
+    return file
+
+
+@pytest.fixture(scope="class")
+def lower_weighting(main_dir) -> pd.DataFrame:
+    """
+    Weighting data to be joined to lower zoning system for testing
+    Returns:
+        pd.DataFrame: Weighting data in a column called 'weight', with
+        an index called 'lower_id', matching lower_id in lower shape.
+    """
+    weighting = pd.DataFrame(
+        data=[
+            10,
+            20,
+            20,
+            30,
+            20,
+            10,
+            10,
+            10,
+            30,
+            20,
+            20,
+            30,
+            30,
+            30,
+            10,
+            10,
+        ],
+        index=range(1, 17),
+        columns=["weight"],
+    )
+    weighting.index.name = "lower_id"
+    file = main_dir / "weighting.csv"
+    weighting.to_csv(main_dir / "weighting.csv")
+    return file
+
+
+@pytest.fixture(scope="class")
+def cache_path(main_dir):
+    path = main_dir / "cache"
+    return path
+
+
+@pytest.fixture(scope="class")
+def out_path(main_dir):
+    path = main_dir / "output"
+    return path
+
+
+@pytest.fixture(scope="class")
+def weighted_config(zone_1_shape, zone_2_shape, lower_zone, lower_weighting,
+                    out_path, cache_path):
+    zone_1 = inputs.ZoneSystemInfo(
+        name="zone_1", shapefile=zone_1_shape, id_col="zone_1_id"
+    )
+    zone_2 = inputs.ZoneSystemInfo(
+        name="zone_2", shapefile=zone_2_shape, id_col="zone_2_id"
+    )
+    lower = inputs.LowerZoneSystemInfo(
+        name="lower_zone",
+        shapefile=lower_zone,
+        id_col="lower_id",
+        weight_data=lower_weighting,
+        data_col="weight",
+        weight_id_col="lower_id",
+    )
+    params = inputs.ZoningTranslationInputs(
+        zone_1=zone_1,
+        zone_2=zone_2,
+        lower_zoning=lower,
+        output_path=out_path,
+        cache_path=cache_path,
+        method="test",
+        tolerance=0.99,
+        rounding=True
+    )
+    return params
+
+
+@pytest.fixture(scope="class")
+def spatial_config(zone_1_shape, zone_2_shape, lower_zone, lower_weighting,
+                   out_path, cache_path):
+    zone_1 = inputs.ZoneSystemInfo(
+        name="zone_1", shapefile=zone_1_shape, id_col="zone_1_id"
+    )
+    zone_2 = inputs.ZoneSystemInfo(
+        name="zone_2", shapefile=zone_2_shape, id_col="zone_2_id"
+    )
+    params = inputs.ZoningTranslationInputs(
+        zone_1=zone_1,
+        zone_2=zone_2,
+        output_path=out_path,
+        cache_path=cache_path,
+        tolerance=0.99,
+        rounding=True
+    )
+    return params
+
+@pytest.fixture(scope="class")
+def dupe_shapes_config(weighted_config):
```

Comment:
This just alters the config so that zone_2 and lower are the same zone system. This used to break the code so I'm including it as a test. I think it would make sense to weight like this if you only wanted a one way translation (i.e. if you only wanted a zone_1 to zone_2 translation it would be ok to weight using zone_2).